client/components/IndividualChallenge.js

```diff
@@ -37,21 +39,22 @@ const _IndividualChallenge = ({
 
   const changeValue = () => {
     // TODO: axios post call to the backend
-    console.log({ html, css, js });
   };
 
-  const updateValue = isSubmit => {
+  const updateValue = (createDiff, isSubmit = false) => {
     setLoading(true);
-    const userAnswer = { html, css, js, submitted: true, challengeId };
-    updateUserchallenge(userAnswer, challengeId, isSubmit)
-      .then(userchallenge => console.log(userchallenge))
+    const userAnswer = { html, css, js, challengeId };
+    updateUserchallenge(userAnswer, userchallenge.id, createDiff, isSubmit)
       .then(() => {
-        setShowModal(isSubmit);
-        setShowCodeMirror(!isSubmit);
+        setShowModal(createDiff);
+        setShowCodeMirror(!createDiff);
         setLoading(false);
+        history.push('/');
```

Comment:
@hgranthorner , here history is undefined.

Reply:
Haoyu, it is not undefined on my end.